### PR TITLE
Fixes #285 and #599

### LINKED ISSUE #285: Update readme with azure sub/role requirements
We got this feedback from a customer in one of the sample repos:

Small suggestion for the Prerequisites section in the README: add a point regarding the fact that the account you will use needs to have access to an Azure subscription (and perhaps even the minimal role required on this subscription).

### PR DESCRIPTION
**Pull request by puicchan**
readme updates.

### CODE REVIEW

**Review by ellismg — APPROVED** [review]
After seeing how many files we had to update, I wonder if we can work towards generating more of these files from templates somehow.  It feels like the there's some consistency here we could tool.  Do you think there's enough sharable content for this to be worthwhile?  Maybe a nice "MQ" exercise?

File: templates/todo/projects/csharp-cosmos-sql/README.md [hunk 1]
```diff
@@ -191,19 +181,7 @@ This template uses [Azure Key Vault](https://docs.microsoft.com/azure/key-vault/
...
-```
-powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/uninstall-azd.ps1' | Invoke-Expression"
-```
-
-Linux/MacOS:
-
-```
-curl -fsSL https://aka.ms/uninstall-azd.sh | bash
-```
+To remove the Azure Developer CLI, refer to [uninstall Azure Developer CLI](https://aka.ms/azd-install?tabs=baremetal%2Cwindows#uninstall-azd).
```
Comment (jongio) [inline templates/todo/projects/csharp-cosmos-sql/README.md:184]: Can we pin to a generic uninstall link or does it have to be windows?

File: templates/todo/projects/csharp-sql/README.md [hunk 2]
```diff
@@ -14,19 +14,9 @@ Let's jump in and get the ToDo app up and running in Azure. When you are finishe
...
-  - Windows:
-    ```powershell
-    powershell -ex AllSigned -c "Invoke-RestMethod 'https://aka.ms/install-azd.ps1' | Invoke-Expression"
-    ```
-  - Linux/MacOS:
-    ```
-    curl -fsSL https://aka.ms/install-azd.sh | bash
-    ```
-- [Azure CLI (2.38.0+)](https://docs.microsoft.com/cli/azure/install-azure-cli)
+- [Azure Developer CLI](https://aka.ms/azd-install)
```
Comment (jongio) [inline templates/todo/projects/csharp-sql/README.md:17]: I would like to follow https://aka.ms/azd/install pattern if possible

**Review by jongio — CHANGES_REQUESTED** [review]

**Comment by jongio** [comment]
> After seeing how many files we had to update, I wonder if we can work towards generating more of these files from templates somehow. It feels like the there's some consistency here we could tool. Do you think there's enough sharable content for this to be worthwhile? Maybe a nice "MQ" exercise?

@ellismg - Yeah, dynamic readmes would be nice, captured: https://github.com/Azure/azure-dev/issues/407

File: templates/todo/projects/csharp-sql/README.md (hunk 2 above)
Comment (puicchan) [reply templates/todo/projects/csharp-sql/README.md:17]: I am reusing an existing aka.ms used in Savannah's blog announcement.

File: templates/todo/projects/csharp-cosmos-sql/README.md (hunk 1 above)
Comment (puicchan) [reply templates/todo/projects/csharp-cosmos-sql/README.md:184]: We don't have an uninstall page in Dev Hub and I have tried without "windows", won't jump to section if not included.

File: templates/todo/projects/nodejs-mongo-swa-func/README.md [hunk 3]
```diff
@@ -14,19 +14,9 @@ Let's jump in and get the ToDo app up and running in Azure. When you are finishe
...
-    ```
-  - Linux/MacOS:
-    ```
-    curl -fsSL https://aka.ms/install-azd.sh | bash
-    ```
-- [Azure CLI (2.38.0+)](https://docs.microsoft.com/cli/azure/install-azure-cli)
+- [Azure Developer CLI](https://aka.ms/azd-install)
 - [Azure Functions Core Tools (4+)](https://docs.microsoft.com/azure/azure-functions/functions-run-local)
 - [Node.js with npm (16.13.1+)](https://nodejs.org/) - for API backend and Web frontend
-- [Git (2.36.1+)](https://git-scm.com/)
```
Comment (vhvb1989) [inline templates/todo/projects/nodejs-mongo-swa-func/README.md:29]: Why is git not a dependency any more?

Comment (puicchan) [reply templates/todo/projects/nodejs-mongo-swa-func/README.md:29]: It's included in the install page on Dev Hub.

**Review by jongio — APPROVED** [review]

**Pull request merged**